src/nvim/ex_getln.c

```diff
@@ -3458,6 +3434,9 @@ void cmdline_ui_flush(void)
     if (line->level == level) {
       if (line->redraw_state == kCmdRedrawAll) {
         ui_ext_cmdline_show(line);
+        if (ui_did_lua_cb) {
+          update_screen();
+        }
```

Comment:
I don't like thisl. it would be better (and generalizable to other usecases) to add whatever `vim.redraw_...()` API to lua so the callback can add redraws as needed.

Reply:
OK yeah sure, if we are OK with exposing `update_screen()` I'll look into a `vim.redraw`.